### DISCUSSION TITLE
MPM-MLS method implementation

### DISCUSSION THREAD

**Discussion by aagrawal05** [ProjectPhysX/FluidX3D#100] [Ideas]
https://www.youtube.com/watch?v=4Y58Pg9tpSo
I feel like something like the above would be a really cool addition to the simulator and would be amazing to see with GPU parallelization with OpenCL. Just a thought.

**Comment by ProjectPhysX** [comment]
You can already do such real-time splashing simulations with FluidX3D!
One example: https://youtu.be/XOfXHgP4jnQ

**Reply by aagrawal05** [reply]
Oh yeah, I didn't see this. That's awesome.
But does it work with rigidbodies? E.g. Can I import a model and have it simulated in the water with gravity? 
On second thought I also realise that this idea is probably outside the scope of the project - isn't a boltzmann lattice method -  but could be a cool project built on top of it. Might try working something out in a fork.

**Reply by ProjectPhysX** [reply]
There is no rigid body floatation mechanism in the software as-is, but it's possible to build this on top purely in the setup script. FluidX3D provides functionality for (moving) object voxelization and for obtaining body forces and torques from the fluid. Combine these in the setup script and you get 2-way coupling with rigid bodies.
